foxtrot-server/src/main/java/com/flipkart/foxtrot/server/resources/Auth.java

```diff
@@ -0,0 +1,142 @@
+import com.flipkart.foxtrot.core.auth.User;
+import com.flipkart.foxtrot.server.auth.AuthConfig;
+import com.flipkart.foxtrot.server.auth.AuthStore;
+import com.flipkart.foxtrot.server.auth.TokenType;
+import com.flipkart.foxtrot.server.auth.io.CreateUserRequest;
+import com.flipkart.foxtrot.server.utils.AuthUtils;
+import io.swagger.annotations.Api;
+import lombok.val;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.Date;
+
+/**
+ *
+ */
+@Path("/v1/auth")
+@Produces(MediaType.APPLICATION_JSON)
+@Api("Auth related APIs. DO NOT expose these for public access.")
+@RolesAllowed(FoxtrotRole.Value.SYSADMIN)
+public class Auth {
+    private final Provider<AuthStore> authProvider;
+    private final AuthConfig authConfig;
+
+    @Inject
+    public Auth(Provider<AuthStore> authProvider, AuthConfig authConfig) {
+        this.authProvider = authProvider;
+        this.authConfig = authConfig;
+    }
+
+    @POST
+    @Path("/users")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response provisionUser(@NotNull @Valid final CreateUserRequest createUserRequest) {
+        val user = new User(createUserRequest.getId(), createUserRequest.getRoles(), createUserRequest.getTables(),
+                            createUserRequest.isSystem(), new Date(), new Date());
+        return Response.ok(authProvider.get().provisionUser(user)).build();
+    }
+
+    @GET
+    @Path("/users/{userId}")
+    public Response getUser(@NotNull @NotEmpty @PathParam("userId") final String userId) {
+        return Response.ok(authProvider.get().getUser(userId)).build();
+    }
+
+    @PUT
+    @Path("/users/{userId}/roles/grant/{role}")
+    public Response grantRole(@NotNull @NotEmpty @PathParam("userId") final String userId,
+                                    @NotNull @PathParam("role") final FoxtrotRole role) {
+        val status = authProvider.get()
+                .grantRole(userId, role);
+        return updateUserResponse(userId, status);
+    }
+
+    @PUT
+    @Path("/users/{userId}/roles/revoke/{role}")
+    public Response revokeRole(@NotNull @NotEmpty @PathParam("userId") final String userId,
+                                    @NotNull @PathParam("role") final FoxtrotRole role) {
+        val status = authProvider.get()
+                .revokeRole(userId, role);
+        return updateUserResponse(userId, status);
+    }
+
+    @PUT
+    @Path("/users/{userId}/tables/access/grant/{table}")
+    public Response grantTableAccess(@NotNull @NotEmpty @PathParam("userId") final String userId,
+                                    @NotNull @NotEmpty @PathParam("table") final String table) {
```

Comment:
Should we expose an api to grant multiple tables in one request? If an installation of foxtrot has multiple tables, it will be lot of work for the admin to give access to each user for multiple tables

Reply:
https://github.com/Flipkart/foxtrot/issues/332